src/index.js

```diff
@@ -1,234 +1,862 @@
+  }
+  if (
+    typeof arguments[1] === 'string' &&
+    arguments[1].includes('/presigned-url?attachment=1')
+  ) {
+    originalResponse.addEventListener('readystatechange', function () {
+      if (originalResponse.readyState === 4) {
+        const jsonBillHref = JSON.parse(originalResponse.responseText).url
+        billsHrefs.push(jsonBillHref)
+      }
+    })
+    return openProxied.apply(this, [].slice.call(arguments))
+  } else {
+    return openProxied.apply(this, [].slice.call(arguments))
+  }
+}
+
+class PayfitContentScript extends ContentScript {
+  addSubmitButtonListener() {
+    // this.log('info', '🤖 addSubmitButtonListener')
+    const formElement = document.querySelector('form')
+    const passwordButton = document.querySelector('._button-login-password')
+    if (passwordButton) {
+      formElement.addEventListener('submit', () => {
+        const email = document.querySelector(
+          '.ulp-authenticator-selector-text'
+        )?.textContent
+        const password = document.querySelector('#password')?.value
+        this.bridge.emit('workerEvent', {
+          event: 'loginSubmit',
+          payload: { email, password }
         })
+      })
+    }
+    const error = document.querySelector('.error')
+    if (error) {
+      this.bridge.emit('workerEvent', {
+        event: 'loginError',
+        payload: { msg: error.innerHTML }
+      })
+    }
+  }
+
+  onWorkerReady() {
+    this.log('info', '🤖 onWorkerReady')
+    if (document.readyState !== 'loading') {
+      this.addSubmitButtonListener.bind(this)()
+    } else {
+      window.addEventListener('DOMContentLoaded', () => {
+        this.log('info', 'DOMLoaded')
+        this.addSubmitButtonListener.bind(this)()
+      })
+    }
+  }
+
+  onWorkerEvent({ event, payload }) {
+    this.log('info', '🤖 onWorkerEvent')
+    if (event === 'loginSubmit') {
+      this.log('info', 'received loginSubmit, blocking user interactions')
+      this.blockWorkerInteractions()
+      const { email, password } = payload || {}
+      if (email && password) {
+        this.log('info', 'Couple email/password found')
+        this.store.userCredentials = { email, password }
+      }
+    } else if (event === 'loginError') {
+      this.log(
+        'info',
+        'received loginError, unblocking user interactions: ' + payload?.msg
+      )
+      this.unblockWorkerInteractions()
+    }
+  }
+
+  async navigateToLoginForm() {
+    this.log('info', '🤖 navigateToLoginForm')
+    await this.goto(baseUrl)
+    await Promise.race([
+      this.waitForElementInWorker('#username'),
+      this.waitForElementInWorker('div[data-testid="userInfoSection"]')
+    ])
+  }
+
+  async ensureAuthenticated({ account }) {
+    // Using a desktop userAgent is mandatory to have access to the user's personnal data
+    await this.bridge.call(
+      'setUserAgent',
+      'Mozilla/5.0 (X11; Ubuntu; Linux x86_64; rv:94.0) Gecko/20100101 Firefox/94.0'
```

Comment:
Maybe you can use a more recent firefox version now. Or better, a chrome version, which will be closer to the webview render engine